js/ui/shellDBus.js

```diff
@@ -534,15 +534,16 @@ const AppStoreService = new Lang.Class({
     },
 
     AddAppIfNotVisible: function(id) {
-        let eventRecorder = EosMetrics.EventRecorder.get_default();
-        let appId = new GLib.Variant('s', id);
-        eventRecorder.record_event(SHELL_APP_ADDED_EVENT, appId);
+        let visibleIcons = IconGridLayout.layout.getIcons(IconGridLayout.DESKTOP_GRID_ID);
+        let isIconVisible = visibleIcons.indexOf(id) != -1;
 
-        if (IconGridLayout.layout.iconIsFolder(id))
-            return;
+        if (!isIconVisible) {
+            let eventRecorder = EosMetrics.EventRecorder.get_default();
+            let appId = new GLib.Variant('s', id);
+            eventRecorder.record_event(SHELL_APP_ADDED_EVENT, appId);
+        }
 
-        let visibleIcons = IconGridLayout.layout.getIcons(IconGridLayout.DESKTOP_GRID_ID);
-        if (visibleIcons.indexOf(id) == -1)
+        if (!IconGridLayout.layout.iconIsFolder(id))
```

Comment:
It probably makes sense to keep this as an early return

Reply:
I don't mind either way. Since there's only one thing to be done in the `!iconIsFolder(id)` case, I kind of liked the more compact style, but I will put the early return back in place.